xarray/coding/strings.py

```diff
@@ -27,11 +28,10 @@ def create_vlen_dtype(element_type):
 
 
 def check_vlen_dtype(dtype):
-    if dtype.kind != "O" or dtype.metadata is None:
+    if is_extension_array_dtype(dtype) or dtype.kind != "O" or dtype.metadata is None:
```

Comment:
These dtype changes are fairly invasive. I'm not sure how Categorical Arrays are ending up in the coding/decoding pipeline. 

In my mind, these would get serialized as netcdf Enum types, and they';d be a new `CategoricalCoder` with `encode` and `decode` methods that handles the transformations appropriately.

Reply:
Yup meant to remove this when I decided not to do i/o a while ago.  We should do that in another PR.